crates/sui-types/src/gas.rs

```diff
@@ -371,6 +371,21 @@ impl<'a> SuiGasStatus<'a> {
     }
 
     pub fn bucketize_computation(&mut self) -> Result<(), ExecutionError> {
+        let computation_cost: u64 = self.gas_used().into();
+        let bucket_cost = get_bucket_cost(&COMPUTATION_BUCKETS, computation_cost);
+        // charge extra on top of `computation_cost` to make the total computation
+        // gas cost a bucket value
+        let extra_charge = bucket_cost.saturating_sub(computation_cost);
+        if extra_charge > 0 {
+            self.deduct_computation_cost(&GasUnits::new(extra_charge).to_unit())
+        } else {
+            // we hit the last bucket and the computation is already more then the
+            // max bucket so just charge as much as it is without buckets
+            Ok(())
+        }
+    }
+
+    pub fn bucketize_computation_legacy(&mut self) -> Result<(), ExecutionError> {
```

Comment:
instead of `legacy` maybe `_v1` to indicate that it is used in version 1? Alternatively just add a comment indicate what protocol versions this is used for.

Reply:
ok I will, to be honest I did not go too deep here because this is not in main, it's just a testnet story until next deployment...